libs/common/src/main/java/org/opensearch/common/StopWatch.java

```diff
@@ -268,4 +267,14 @@ public TimeValue getTime() {
         }
     }
 
+    /**
+     * Stops the watch and auto calls close in try-with-resources usage
+     *
+     * @opensearch.internal
+     */
+    public interface TimingHandle extends AutoCloseable {
```

Comment:
The `Releasable` has subtle but very important detail regarding exceptions:

```
/**
 * Specialization of {@link AutoCloseable} that may only throw an {@code BaseOpenSearchException}.
 *
 */
public interface Releasable extends Closeable {
  ...
}

```

Should we assume the same? Or follow just inherit the exception semantics from `AutoCloseable`?

```
public interface TimingHandle extends AutoCloseable {
}
```

Reply:
@andrross correct, that just a comment but does it apply to `TimingHandle` ? Any reasons we need to override `AutoCloseable::close` to strip the `throws Exception` from the method signature?

Reply:
> The only difference between AutoCloseable and Releasable is that Releasable cannot throw a checked exception.

Technically `Releasable extends Closeable` which downscopes the `::close` method to only throw `IOException`. So if the goal is to decouple `Releasable` from `StopWatch` so we can refactor it out of `opensearch-core` to `opensearch-common` then `StopWatch::timing` could return a `Closeable` instead of a new `TimingHandle`.

> Any reasons we need to override AutoCloseable::close to strip the throws Exception from the method signature?

In core, the impact of refactoring `::close` back to throwing any Exception (`IOException` if we decide to return a `Closeable` instead of a `TimingHandle`) only percolates up to private `ClusterApplierService::applyChanges` which is called from `ClusterApplierService::runTask`. The try-with-resources will throw the Exception that is ultimately caught in the try-catch of `runTask`.  So, to answer your question, we could technically do away with all this and percolate the `IOException` of `Closeable` up to `runTask`. This would be breaking, however, for any downstream users of `StopWatch::timing`, they'd not only have to refactor to the new type (`Closeable`), which they're going to have to do anyway if we keep `TimingHandle`, they would also have to add a try-catch to the method call or percolate the exception up their own stack. 

Stepping back to not lose the Forest for the Trees here, my intent of these refactor PRs is to be minimally destructive to the logic path that enables a full decoupling from `:server`. So we could make a decision now to not strip the `Exception` (or just throw the `IOException` from `Closeable`) and leave the current logic alone (only requiring downstreams to refactor from `Releaseable` to `Closeable`) or be less destructive and keep the `TimingHandle` so downstream doesn't have to add Exception handling logic.

I vote the latter. Just strip the Exception since it's already not thrown in the current version.